tmt/package_managers/apk.py

```diff
@@ -0,0 +1,164 @@
+            self._sudo_prefix = Command('sudo')
+
+        else:
+            self._sudo_prefix = Command()
+
+        command = Command()
+
+        command += self._sudo_prefix
+        command += Command('apk')
+
+        return (command, Command())
+
+    def path_to_package(self, path: FileSystemPath) -> Package:
+        """
+        Find a package providing given filesystem path.
+
+        This is not easily possible in Alpine. There is `apk-file` utility
+        available but it seems unrealiable. Support only a fixed set
+        of mappings until a better solution is available.
+        """
+
+        string_path = str(path)
+
+        if string_path in PACKAGE_PATH:
+            return Package(PACKAGE_PATH[string_path])
+
+        raise GeneralError(f"Unsupported package path '{path} for Alpine Linux.")
+
+    def _reduce_to_packages(self, *installables: Installable) -> ReducedPackages:
+        packages: ReducedPackages = []
+
+        for installable in installables:
+            if isinstance(installable, (Package, PackagePath)):
+                packages.append(installable)
+
+            elif isinstance(installable, FileSystemPath):
+                packages.append(self.path_to_package(installable))
+
+            else:
+                raise GeneralError(f"Package specification '{installable}' is not supported.")
+
+        return packages
+
+    def _construct_presence_script(
+            self, *installables: Installable) -> tuple[ReducedPackages, ShellScript]:
+        reduced_packages = self._reduce_to_packages(*installables)
+
+        shell_script = ShellScript(
+            f'apk info -e {" ".join(escape_installables(*reduced_packages))}')
+
+        return reduced_packages, shell_script
+
+    def check_presence(self, *installables: Installable) -> dict[Installable, bool]:
+        reduced_packages, presence_script = self._construct_presence_script(*installables)
+
+        try:
+            output = self.guest.execute(presence_script)
+            stdout, stderr = output.stdout, output.stderr
+
+        except RunError as exc:
+            stdout, stderr = exc.stdout, exc.stderr
+
+        if stdout is None or stderr is None:
+            raise GeneralError("apk presence check output is None")
+
+        results: dict[Installable, bool] = {}
+
+        for installable, package in zip(installables, reduced_packages):
+            match = re.search(rf'^{re.escape(str(package))}\s', stdout)
+
+            if match is not None:
+                results[installable] = True
+                continue
+
+            results[installable] = False
+
+        return results
+
+    def install(
+            self,
+            *installables: Installable,
+            options: Optional[Options] = None) -> CommandOutput:
+        options = options or Options()
+
+        packages = self._reduce_to_packages(*installables)
+
+        script = ShellScript(
+            f'{self.command.to_script()} add '
```

Comment:
Should `install_command` be used here if we have it defined above? Or remove it above as it is not used?

Reply:
@psss

ack on `install_command`

```
I guess the image is needed because we rely on bash for all the shell script operations? The other two packages though could/should be removed, I'd say.
```

Well, this is a testing container for solely test purposes. I am arguing that to make the tests more unified with the rest and thus less confusing.

Reply:
install command was addressed, the discussion with `tar` and `flock` is open.

If we add need to add `bash`, I would keep it for sanity of the unit tests.

Reply:
talked to @happz privately, I will remove `tar` and `flock` we can use something else